packages/protocol/contracts-0.8/common/MintGoldSchedule.sol

```diff
@@ -0,0 +1,297 @@
+    setCommunityRewardFraction(_communityRewardFraction);
+    setCarbonOffsettingFund(_carbonOffsettingPartner, _carbonOffsettingFraction);
+  }
+
+  /**
+   * @notice Mints CELO to the community and carbon offsetting funds according to the predefined schedule.
+   */
+  function mintAccordingToSchedule() external nonReentrant onlyL2 returns (bool) {
+    (
+      uint256 targetGoldTotalSupply,
+      uint256 communityRewardFundMintAmount,
+      uint256 carbonOffsettingPartnerMintAmount
+    ) = getTargetGoldTotalSupply();
+
+    uint256 mintableAmount = targetGoldTotalSupply - getGoldToken().totalSupply();
+
+    require(mintableAmount > 0, "Mintable amount must be greater than zero");
+    require(
+      getRemainingBalanceToMint() >= mintableAmount,
+      "Insufficient unlocked balance to mint amount"
+    );
+    totalMintedBySchedule = totalMintedBySchedule + (mintableAmount);
+
+    IGoldToken goldToken = IGoldToken(address(getGoldToken()));
+    require(
+      goldToken.mint(communityRewardFund, communityRewardFundMintAmount),
+      "Failed to mint to community partner."
+    );
+
+    require(
+      goldToken.mint(carbonOffsettingPartner, carbonOffsettingPartnerMintAmount),
+      "Failed to mint to carbon offsetting partner."
+    );
+    return true;
+  }
+
+  /**
+   * @notice Returns the community reward fraction.
+   * @return The percentage of total reward which goes to the community funds.
+   */
+  function getCommunityRewardFraction() external view returns (uint256) {
+    return communityRewardFraction.unwrap();
+  }
+
+  /**
+   * @notice Returns the carbon offsetting partner reward fraction.
+   * @return The percentage of total reward which goes to the carbon offsetting partner.
+   */
+  function getCarbonOffsettingFraction() external view returns (uint256) {
+    return carbonOffsettingFraction.unwrap();
+  }
+
+  /**
+   * @notice Returns the storage, major, minor, and patch version of the contract.
+   * @return Storage version of the contract.
+   * @return Major version of the contract.
+   * @return Minor version of the contract.
+   * @return Patch version of the contract.
+   */
+  function getVersionNumber() external pure returns (uint256, uint256, uint256, uint256) {
+    return (1, 1, 0, 0);
+  }
+
+  /**
+   * @notice Sets the community reward percentage
+   * @param value The percentage of the total reward to be sent to the community funds.
+   * @return True upon success.
+   */
+  function setCommunityRewardFraction(
+    uint256 value
+  ) public onlyOwner whenDependenciesSet returns (bool) {
+    uint256 timeSinceL2 = block.timestamp - l2StartTime;
+    uint256 linearSecondsLeft = SECONDS_LINEAR - (l2StartTime - genesisStartTime);
+
+    require(
+      timeSinceL2 < linearSecondsLeft,
+      "Can only update fraction once block reward calculation for years 15-30 has been implemented."
+    );
+    require(
+      value != communityRewardFraction.unwrap() && value < FixidityLib.fixed1().unwrap(),
```

Comment:
Rather than unwrapping communityRewardFraction and fixed1, can we wrap value (since we have to wrap it later anyways) and compare it wrapped ?